ShowMyCursedRealm/room.aadl

```diff
@@ -1,45 +1,94 @@
 package room
 public
+with threads;
+with deployment;
+	
 	
 	device camera
 		features
-			video: out data port;
-			camera_on: in event port;
+			video_stream: out data port threads::Video_Stream;
+			camera_on_off: in event port;
 	end camera;
 	
 	device movement_sensor
 		features
-			movement_detected: out event port;
+			movement_data: out data port threads::BooleanValue;
 	end movement_sensor;
 	
 	device pin_lock
 		features
-			entered_pin: out data port;
+			entered_pin: out data port threads::Pin;
 	end pin_lock;	
 	
-	device open_sensor
+	device door_sensor
 		features
-			object_opened: out event port;
-	end open_sensor;
-	
+			state: out data port threads::Door_State;
+	end door_sensor;
+	device window_sensor
+		features
+			state: out data port threads::Window_State;
+	end window_sensor;
```

Comment:
W sumie sensowny podział, bo te urządzenia pewnie mogą być reprezentowane przez różne obiekty fizyczne i dobrze jest to podkreślić.